src/bulkExport/bulkExportResults.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import AWS from 'aws-sdk';
+
+const EXPIRATION_TIME_SECONDS = 1800;
+const EXPORT_RESULTS_BUCKET = process.env.EXPORT_RESULTS_BUCKET || ' ';
+const EXPORT_RESULTS_SIGNER_ROLE_ARN = process.env.EXPORT_RESULTS_SIGNER_ROLE_ARN || '';
+
+const getFiles = async (jobId: string): Promise<string[]> => {
+    const s3 = new AWS.S3();
+
+    const listObjecsResult = await s3.listObjectsV2({ Bucket: EXPORT_RESULTS_BUCKET, Prefix: jobId }).promise();
+    return listObjecsResult.Contents!.map(x => x.Key!);
+};
+
+const signExportResults = async (keys: string[]): Promise<{ key: string; url: string }[]> => {
+    if (keys.length === 0) {
+        return [];
+    }
+    const sts = new AWS.STS();
+    const assumeRoleResponse = await sts
+        .assumeRole({
+            RoleArn: EXPORT_RESULTS_SIGNER_ROLE_ARN,
+            RoleSessionName: 'signBulkExportResults',
+            DurationSeconds: EXPIRATION_TIME_SECONDS,
+        })
+        .promise();
+
+    const s3 = new AWS.S3({
+        credentials: {
+            accessKeyId: assumeRoleResponse.Credentials!.AccessKeyId,
+            secretAccessKey: assumeRoleResponse.Credentials!.SecretAccessKey,
+            sessionToken: assumeRoleResponse.Credentials!.SessionToken,
+        },
+    });
+
+    return Promise.all(
+        keys.map(async key => ({
+            key,
+            url: await s3.getSignedUrlPromise('getObject', {
+                Bucket: EXPORT_RESULTS_BUCKET,
+                Key: key,
+                Expires: EXPIRATION_TIME_SECONDS,
+            }),
+        })),
+    );
+};
+
+const getResourceType = (key: string, jobId: string): string => {
+    const regex = new RegExp(`^${jobId}/([A-Za-z]+)-.*$`);
```

Comment:
I am not sure how the files are being created but is the `.*` prescriptive enough? In your PR message it appears it may just be a number so perhaps you want to use `\d+`

Reply:
I just updated the regex. 

yeah, I left it vague since we weren't 100% sure about how glue would name the files. and I just needed to parse the resourcetype. Now that we know that we are manually renaming the files at the end of the glue job we can be more specific about it.